### ISSUE TITLE
network connections not detected. Unable to reach google.com...

### ISSUE THREAD

**Issue by uitta** [Varying-Vagrant-Vagrants/VVV#1585]
I've read every (hope so) thread about it but still i've got some issues with the connection while tryin `vagrant up` 
I've got two different wireless networks, and using one of them i can make vvv works but on the other it says
`  default: Network connection not detected. Unable to reach google.com...
default: No network connection available, skipping package installation
`
and than the process stops

On the working one the ip address i get is 192.168.0.100 (also the router is set to use a vpn) and on the other one is 192.168.1.103, so none of them is in the 192.168.50.x subnet
Is there something i am missing about it?
I tried using
`config.vm.network :private_network, id: "vvv_primary", type:'dhcp'`
but still no luck.

I know, i should use only the one that works but still sometimes i have to use the one that doesn't work.
Is there any suggestion why it act this way?
Thanks a lot

**Comment by tomjn** [comment]
The IP of the VVV instance isn't relevant, the problem is which network adapter to map on to. I'm afraid if you're constantly switching network adapters then you'll be constantly switching VVV.

When you say you have multiple wireless networks, do you mean you have a singular wifi adaptor that connects to 2 separate wifi networks ( aka different SSID, different names etc? )

Or do you mean you have the same wifi network, but you have multiple Wifi chips/network adaptors?

> On the working one the ip address i get is 192.168.0.100 (also the router is set to use a vpn) and on the other one is 192.168.1.103, so none of them is in the 192.168.50.x subnet

How are you identifying the IP here? Is this the IP reported inside the VM? Note that VVVs IP internally is for a private internal network, it isn't an IP on your wifi LAN. VVVs IP is always `192.168.50.4` as defined in the vagrant file unless overriden using `private_network_ip` in the `vvv-custom.yml` file

**Comment by tomjn** [comment]
Additionally, the error you mentioned puts out a large amount of debugging info that would be helpful, and the issue template asks you to include your vagrant version and other information about VBox and VVV, can you include this? **We ask for it because it's super important, but your issue doesn't include that information**

**Comment by uitta** [comment]
Thanks for your reply.
First of all, i've got one wireless network from a router (one SSID and it's the one i cannot work with) and to the same router i plugged in another router for the second wireless network (because i'm using it with a vpn and it's another SSID). 
Also the IP i'm talkin about is the host machine (my computer), because i thaught that was the important IP.  I'm sort of a beginner with vagrant and vvv and i'm trying to learn how to use it.
I just installed everything from scratch, i'm using 2.1.2 for Vagrant, 2.2.1 for vvv and 5.2.14 for VBOX (really sorry for the missing info).

and this is all the log
`__ __ __ __ 
\ V\ V\ V / Varying Vagrant Vagrants
 \_/\_/\_/  v2.2.1-git::master
 
Docs:       https://varyingvagrantvagrants.org/
Contribute: https://github.com/varying-vagrant-vagrants/vvv
Dashboard:  http://vvv.test

IMPORTANT: Copying vvv-config.yml to vvv-custom.yml, make all modifications to vvv-custom.yml in future

Bringing machine 'default' up with 'virtualbox' provider...
==> default: Importing base box 'ubuntu/trusty64'...
==> default: Matching MAC address for NAT networking...
==> default: Checking if box 'ubuntu/trusty64' is up to date...
==> default: A newer version of the box 'ubuntu/trusty64' for provider 'virtualbox' is
==> default: available! You currently have version '20180709.0.1'. The latest is version
==> default: '20180709.0.4'. Run `vagrant box update` to update.
==> default: Setting the name of the VM: wpdev_3163b162781
==> default: Clearing any previously set forwarded ports...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
    default: Adapter 2: hostonly
==> default: Forwarding ports...
    default: 22 (guest) => 2222 (host) (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection reset. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Connection reset. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Connection reset. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Connection reset. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Connection reset. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Connection reset. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Connection reset. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Connection reset. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Connection reset. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Connection reset. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Connection reset. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Connection reset. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Connection reset. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: 
    default: Vagrant insecure key detected. Vagrant will automatically replace
    default: this with a newly generated keypair for better security.
    default: 
    default: Inserting generated public key within guest...
    default: Removing insecure key from the guest if it's present...
    default: Key inserted! Disconnecting and reconnecting using new SSH key...
==> default: Machine booted and ready!
==> default: Checking for guest additions in VM...
    default: The guest additions on this VM do not match the installed version of
    default: VirtualBox! In most cases this is fine, but in rare cases it can
    default: prevent things such as shared folders from working properly. If you see
    default: shared folder errors, please make sure the guest additions within the
    default: virtual machine match the version of VirtualBox you have installed on
    default: your host and reload your VM.
    default: 
    default: Guest Additions Version: 4.3.36
    default: VirtualBox Version: 5.2
==> default: [vagrant-hostsupdater] Checking for host entries
==> default: [vagrant-hostsupdater]   found entry for: 192.168.50.4 vvv
==> default: [vagrant-hostsupdater]   found entry for: 192.168.50.4 vvv.dev
==> default: [vagrant-hostsupdater]   found entry for: 192.168.50.4 vvv.test
==> default: [vagrant-hostsupdater]   found entry for: 192.168.50.4 vvv.local
==> default: [vagrant-hostsupdater]   found entry for: 192.168.50.4 vvv.localhost
==> default: [vagrant-hostsupdater]   found entry for: 192.168.50.4 local.wordpress.test
==> default: [vagrant-hostsupdater]   found entry for: 192.168.50.4 src.wordpress-develop.test
==> default: [vagrant-hostsupdater]   found entry for: 192.168.50.4 build.wordpress-develop.test
==> default: Setting hostname...
==> default: Configuring and enabling network interfaces...
==> default: Mounting shared folders...
    default: /var/log => /Users/vitta/wpdev/log
    default: /srv/www => /Users/vitta/wpdev/www
    default: /vagrant => /Users/vitta/wpdev
    default: /srv/config => /Users/vitta/wpdev/config
    default: /srv/database => /Users/vitta/wpdev/database
==> default: Running provisioner: fix-no-tty (shell)...
    default: Running: inline script
==> default: Running provisioner: default (shell)...
    default: Running: /var/folders/5f/np90kvb91999jyc5370jzt2m0000gn/T/vagrant-shell20180716-1072-1xpgi6z.sh
    default: Network connection not detected. Unable to reach google.com...
    default: No network connection available, skipping package installation
==> default: Running provisioner: dashboard (shell)...
    default: Running: /var/folders/5f/np90kvb91999jyc5370jzt2m0000gn/T/vagrant-shell20180716-1072-1rx3sfw.sh
    default: Downloading dashboard, see https://github.com/Varying-Vagrant-Vagrants/dashboard.git
    default: /tmp/vagrant-shell: line 11: git: command not found
    default: /tmp/vagrant-shell: line 12: cd: /srv/www/default/dashboard: No such file or directory
    default: /tmp/vagrant-shell: line 13: git: command not found
==> default: Running provisioner: utility-source-core (shell)...
    default: Running: /var/folders/5f/np90kvb91999jyc5370jzt2m0000gn/T/vagrant-shell20180716-1072-fyam1q.sh
    default: Downloading core resources, see https://github.com/Varying-Vagrant-Vagrants/vvv-utilities.git
    default: sudo: git: command not found
    default: /tmp/vagrant-shell: line 17: cd: /vagrant/provision/resources/core: No such file or directory
    default: sudo: git: command not found
==> default: Running provisioner: utility-core-memcached-admin (shell)...
    default: Running: /var/folders/5f/np90kvb91999jyc5370jzt2m0000gn/T/vagrant-shell20180716-1072-y19oe5.sh
==> default: Running provisioner: utility-core-opcache-status (shell)...
    default: Running: /var/folders/5f/np90kvb91999jyc5370jzt2m0000gn/T/vagrant-shell20180716-1072-1p6jeum.sh
==> default: Running provisioner: utility-core-phpmyadmin (shell)...
    default: Running: /var/folders/5f/np90kvb91999jyc5370jzt2m0000gn/T/vagrant-shell20180716-1072-vrczrm.sh
==> default: Running provisioner: utility-core-webgrind (shell)...
    default: Running: /var/folders/5f/np90kvb91999jyc5370jzt2m0000gn/T/vagrant-shell20180716-1072-7jlpf5.sh
==> default: Running provisioner: utility-core-trusted-hosts (shell)...
    default: Running: /var/folders/5f/np90kvb91999jyc5370jzt2m0000gn/T/vagrant-shell20180716-1072-ffw83a.sh
==> default: Running provisioner: utility-core-tls-ca (shell)...
    default: Running: /var/folders/5f/np90kvb91999jyc5370jzt2m0000gn/T/vagrant-shell20180716-1072-v38qyh.sh
==> default: Running provisioner: site-wordpress-default (shell)...
    default: Running: /var/folders/5f/np90kvb91999jyc5370jzt2m0000gn/T/vagrant-shell20180716-1072-1z02izz.sh
    default: Downloading wordpress-default, see https://github.com/Varying-Vagrant-Vagrants/custom-site-template.git
    default: sudo: git: command not found
    default: find: 
    default: `/srv/www/wordpress-default'
    default: : No such file or directory
    default: nginx: unrecognized service
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.`

Thanks and sorry again if i'm missing something or, what i think is possible, if i'm making some huge mistake due to my ignorance.

**Comment by tomjn** [comment]
Can you run using the `develop` branch instead?

**Comment by uitta** [comment]
The network with the vpn is the one that actually works.
The main one is the one that i cannot work with. Also it's a router/gateway from my isp (FASTWEB, italian) and i didn't change any setting (apart from the SSID name and password)
This is the log from branch.

``__ __ __ __ 
\ V\ V\ V / Varying Vagrant Vagrants
 \_/\_/\_/  v2.2.2-develop-git::develop

Platform:   darwin13.4.0 shell:/bin/bash
Vagrant:    2.1.2
VirtualBox: 5.2.14r123301

Docs:       https://varyingvagrantvagrants.org/
Contribute: https://github.com/varying-vagrant-vagrants/vvv
Dashboard:  http://vvv.test

Bringing machine 'default' up with 'virtualbox' provider...
==> default: Importing base box 'ubuntu/trusty64'...
==> default: Matching MAC address for NAT networking...
==> default: Checking if box 'ubuntu/trusty64' is up to date...
==> default: A newer version of the box 'ubuntu/trusty64' for provider 'virtualbox' is
==> default: available! You currently have version '20180709.0.1'. The latest is version
==> default: '20180709.0.4'. Run `vagrant box update` to update.
==> default: Setting the name of the VM: wpdev_3163b162781
==> default: Clearing any previously set forwarded ports...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
    default: Adapter 2: hostonly
==> default: Forwarding ports...
    default: 22 (guest) => 2222 (host) (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection reset. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Connection reset. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Connection reset. Retrying...
==> default: Machine booted and ready!
==> default: Checking for guest additions in VM...
    default: The guest additions on this VM do not match the installed version of
    default: VirtualBox! In most cases this is fine, but in rare cases it can
    default: prevent things such as shared folders from working properly. If you see
    default: shared folder errors, please make sure the guest additions within the
    default: virtual machine match the version of VirtualBox you have installed on
    default: your host and reload your VM.
    default: 
    default: Guest Additions Version: 4.3.36
    default: VirtualBox Version: 5.2
==> default: [vagrant-hostsupdater] Checking for host entries
==> default: [vagrant-hostsupdater]   found entry for: 192.168.50.4 vvv
==> default: [vagrant-hostsupdater]   found entry for: 192.168.50.4 vvv.dev
==> default: [vagrant-hostsupdater]   found entry for: 192.168.50.4 vvv.test
==> default: [vagrant-hostsupdater]   found entry for: 192.168.50.4 vvv.local
==> default: [vagrant-hostsupdater]   found entry for: 192.168.50.4 vvv.localhost
==> default: [vagrant-hostsupdater]   found entry for: 192.168.50.4 local.wordpress.test
==> default: [vagrant-hostsupdater]   found entry for: 192.168.50.4 src.wordpress-develop.test
==> default: [vagrant-hostsupdater]   found entry for: 192.168.50.4 build.wordpress-develop.test
==> default: Setting hostname...
==> default: Configuring and enabling network interfaces...
==> default: Mounting shared folders...
    default: /var/log => /Users/vitta/wpdev/log
    default: /srv/www => /Users/vitta/wpdev/www
    default: /vagrant => /Users/vitta/wpdev
    default: /srv/config => /Users/vitta/wpdev/config
    default: /srv/database => /Users/vitta/wpdev/database
==> default: Running provisioner: fix-no-tty (shell)...
    default: Running: inline script
==> default: Running provisioner: default (shell)...
    default: Running: /var/folders/5f/np90kvb91999jyc5370jzt2m0000gn/T/vagrant-shell20180716-1559-1neyo5p.sh
    default: Network connection not detected. Unable to reach google.com...
    default:  
    default: #################################################################
    default:  
    default: Problem:
    default:  
    default: Provisioning needs a network connection but none was found.
    default: VVV tried to ping google.com, and got no response.
    default:  
    default: Make sure you have a working internet connection, that you 
    default: restarted after installing VirtualBox and Vagrant, and that 
    default: they aren't blocked by a firewall or security software. If
    default: you can load https://google.com in your browser, then VVV
    default: should be able to connect.
    default:  
    default: Also note that some users have reported issues when combined
    default: with VPNs, disable your VPN and reprovision to see if this is
    default: the cause.
    default:  
    default: Additionally, if you're at a contributor day event, be kind,
    default: provisioning involves downloading things, a full provision may 
    default: ruin the wifi for everybody else :(
    default:  
    default: Network ifconfig output:
    default:  
    default: eth0      Link encap:Ethernet  HWaddr 08:00:27:15:1e:a8  
    default:           inet addr:10.0.2.15  Bcast:10.0.2.255  Mask:255.255.255.0
    default:           inet6 addr: fe80::a00:27ff:fe15:1ea8/64 Scope:Link
    default:           UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
    default:           RX packets:1836 errors:0 dropped:0 overruns:0 frame:0
    default:           TX packets:1372 errors:0 dropped:0 overruns:0 carrier:0
    default:           collisions:0 txqueuelen:1000 
    default:           RX bytes:215013 (215.0 KB)  TX bytes:167993 (167.9 KB)
    default: 
    default: eth1      Link encap:Ethernet  HWaddr 08:00:27:b4:15:4c  
    default:           inet addr:192.168.50.4  Bcast:192.168.50.255  Mask:255.255.255.0
    default:           UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
    default:           RX packets:2 errors:0 dropped:0 overruns:0 frame:0
    default:           TX packets:6 errors:0 dropped:0 overruns:0 carrier:0
    default:           collisions:0 txqueuelen:1000 
    default:           RX bytes:1180 (1.1 KB)  TX bytes:1044 (1.0 KB)
    default: 
    default: lo        Link encap:Local Loopback  
    default:           inet addr:127.0.0.1  Mask:255.0.0.0
    default:           inet6 addr: ::1/128 Scope:Host
    default:           UP LOOPBACK RUNNING  MTU:65536  Metric:1
    default:           RX packets:0 errors:0 dropped:0 overruns:0 frame:0
    default:           TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
    default:           collisions:0 txqueuelen:0 
    default:           RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)
    default:  
    default: No network connection available, aborting provision. Try 
    default: provisioning again once network connectivity is restored.
    default: If that doesn't work, and you're sure you have a strong 
    default: internet connection, open an issue on GitHub, and include the 
    default: output above so that the problem can be debugged
    default:  
    default: vagrant reload --provision
    default:  
    default: https://github.com/Varying-Vagrant-Vagrants/VVV/issues
    default:  
    default: #################################################################
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.``

**Comment by uitta** [comment]
also i've got no firewall.

**Comment by tomjn** [comment]
@uitta the domain tests was changed to launchpad recently, you may get different results.

However, I am unable to explain why it works with one network but not with the other. To clarify things further, I present a matrix of tried and untried combinations:


|Network|with VPN|without VPN|
|-|-|-|
|Fastweb|?|🚫|
|wifi 2| ✅| ? | 

As an aside, could you not just provision on the one that works? You don't need an internet connection to run `vagrant up` or `vagrant halt`, only `vagrant up --provision` or `vagrant provision` need a connection

**Comment by uitta** [comment]
Thanks for your reply, i didn’t tried every combination, because if i use the vpn on the first router, it will be used from the second router as well (i think). 
But thanks for the last suggestion, that was something i didn’t thought about; so you say i can make the first vagrant up on the wireless that works, and then,unless i need to provision it again, work either on the wireless that didn’t work?

**Comment by tomjn** [comment]
`vagrant up` just turns the VM on, if you've already done provisioning then a simple `vagrant up` should work with your computer in airplane mode with all network interfaces turned off.

Provisioning on the other hand requires a network connection. Think of it as installing a device vs using it. Provisioning is going to the store and buying a new toaster, unpacking it, finding a spare socket, etc. Vagrant up is just toggling the on/off button.

**Comment by bigmazi** [comment]
Hi, I have yet not install vvv but sounds really interesting so for sure I will do this.
Back to your problem please maybe check:
1. on network that working trace route to google by command line put on linux traceroute google.com
2. on network that not working the same 

Till you are answer maybe I will install vvv and may say more about this.

**Comment by tomjn** [comment]
Note that on the develop branch it pokes the Ubuntu launchpad domain
instead of Google. There were issues involving Chinese users and those in
other countries that block google

**Comment by tomjn** [comment]
@uitta is this still an issue for you on the `develop` branch?

**Comment by tomjn** [comment]
Closing this due to inactivity, the original issue of pinging google.com isn't a thing as v2.3 doesn't ping google
